### PR TITLE
added pagination, sorting and swagger to my project

### PR DESCRIPTION
**Pull request by Nazar090**

### CODE REVIEW

File: src/main/java/com/example/onlinebookstore/controller/BookController.java [hunk 1]
```diff
@@ -29,6 +29,7 @@ public class BookController {
     public List<BookDto> getAll(Pageable pageable) {
         return bookService.findAll(pageable);
     }
+    // I added pagination, sorting and swagger(but I did it before finishing previous task)
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/onlinebookstore/controller/BookController.java]: Where are Swagger annotations?

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]

**Review by Elena-Bruyako — APPROVED** [review]

File: src/main/java/com/example/onlinebookstore/controller/AuthController.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+import com.example.onlinebookstore.service.UserService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RequiredArgsConstructor
+@RestController
+@RequestMapping("/auth")
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/onlinebookstore/controller/AuthController.java:18]: swagger annotations are missing

File: src/main/java/com/example/onlinebookstore/controller/AuthController.java [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+public class AuthController {
+    private final UserService userService;
+
+    @PostMapping("/registration")
+    public UserResponseDto registerUser(@RequestBody UserRegistrationRequestDto requestDto)
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/onlinebookstore/controller/AuthController.java]:
```suggestion
    public UserResponseDto registerUser(@RequestBody @Valid UserRegistrationRequestDto requestDto)
```

File: src/main/java/com/example/onlinebookstore/model/User.java [hunk 4]
```diff
@@ -1,4 +1,39 @@
...
+@Getter
+@Setter
+@SQLDelete(sql = "UPDATE employees SET is_deleted = true WHERE id=?")
+@SQLRestriction(value = "is_deleted=false")
+@Table(name = "users")
 public class User {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    @Column(nullable = false)
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/onlinebookstore/model/User.java]:
```suggestion
    @Column(nullable = false, unique = true)
```

File: src/main/java/com/example/onlinebookstore/model/User.java [hunk 5]
```diff
@@ -1,4 +1,39 @@
...
+    private String email;
+    @Column(nullable = false)
+    private String password;
+    @JsonProperty("first_name")
+    @Column(name = "first_name", nullable = false)
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/onlinebookstore/model/User.java]:
```suggestion
    @Column(nullable = false)
```

File: src/main/java/com/example/onlinebookstore/model/User.java [hunk 6]
```diff
@@ -1,4 +1,39 @@
...
+    private String firstName;
+    @JsonProperty("last_name")
+    @Column(name = "last_name", nullable = false)
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/onlinebookstore/model/User.java]:
```suggestion
    @Column(nullable = false)
```

File: src/main/java/com/example/onlinebookstore/model/User.java [hunk 7]
```diff
@@ -1,4 +1,39 @@
...
+@Table(name = "users")
 public class User {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    @Column(nullable = false)
+    private String email;
+    @Column(nullable = false)
+    private String password;
+    @JsonProperty("first_name")
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/onlinebookstore/model/User.java]:
```suggestion
```

File: src/main/java/com/example/onlinebookstore/model/User.java [hunk 8]
```diff
@@ -1,4 +1,39 @@
...
+    @Column(name = "first_name", nullable = false)
+    private String firstName;
+    @JsonProperty("last_name")
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/onlinebookstore/model/User.java]:
```suggestion
```

File: src/main/java/com/example/onlinebookstore/model/User.java [hunk 9]
```diff
@@ -1,4 +1,39 @@
...
+    @Column(name = "last_name", nullable = false)
+    private String lastName;
+    @JsonProperty("shipping_address")
+    @Column(name = "shipping_address", nullable = false)
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/onlinebookstore/model/User.java]:
```suggestion
```

File: src/main/java/com/example/onlinebookstore/service/UserServiceImpl.java [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+@Service
+@RequiredArgsConstructor
+public class UserServiceImpl implements UserService {
+    private final UserRepository userRepository;
+    private final UserMapper userMapper;
+
+    @Override
+    public UserResponseDto register(UserRegistrationRequestDto requestDto)
+            throws RegistrationException {
+        if (userRepository.findByEmail(requestDto.getEmail()).isPresent()) {
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/onlinebookstore/service/UserServiceImpl.java]:
```suggestion
        if (userRepository.existsByEmail(requestDto.getEmail())) {
```

File: src/main/java/com/example/onlinebookstore/service/UserServiceImpl.java [hunk 11]
```diff
@@ -0,0 +1,33 @@
...
+            throw new RegistrationException("Can't register user");
+        }
+        User user = new User();
+        user.setEmail(requestDto.getEmail());
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/onlinebookstore/service/UserServiceImpl.java]: use mapper

File: src/main/java/com/example/onlinebookstore/service/UserServiceImpl.java [hunk 12]
```diff
@@ -0,0 +1,33 @@
...
+@RequiredArgsConstructor
+public class UserServiceImpl implements UserService {
+    private final UserRepository userRepository;
+    private final UserMapper userMapper;
+
+    @Override
+    public UserResponseDto register(UserRegistrationRequestDto requestDto)
+            throws RegistrationException {
+        if (userRepository.findByEmail(requestDto.getEmail()).isPresent()) {
+            throw new RegistrationException("Can't register user");
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/onlinebookstore/service/UserServiceImpl.java]: wrong message, user already exists

File: src/main/java/com/example/onlinebookstore/service/UserServiceImpl.java [hunk 13]
```diff
@@ -0,0 +1,33 @@
...
+        }
+        User user = new User();
+        user.setEmail(requestDto.getEmail());
+        user.setPassword(requestDto.getPassword());
+        user.setFirstName(requestDto.getFirstName());
+        user.setLastName(requestDto.getLastName());
+        user.setShippingAddress(requestDto.getShippingAddress());
+        User savedUser = userRepository.save(user);
+        return userMapper.toDo(savedUser);
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/onlinebookstore/service/UserServiceImpl.java]:
```suggestion
        userRepository.save(user);
        return userMapper.toDo(user);
```

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
see pr

File: src/main/java/com/example/onlinebookstore/model/User.java (hunk 7 above)
Comment (Elena-Bruyako) [reply src/main/java/com/example/onlinebookstore/model/User.java]: Why do you need JsonProperty annotation?

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
see comment

File: src/main/java/com/example/onlinebookstore/model/User.java [hunk 14]
```diff
@@ -1,4 +1,34 @@
...
+import lombok.Getter;
+import lombok.Setter;
+import org.hibernate.annotations.SQLDelete;
+import org.hibernate.annotations.SQLRestriction;
+
+@Entity
+@Getter
+@Setter
+@SQLDelete(sql = "UPDATE employees SET is_deleted = true WHERE id=?")
+@SQLRestriction(value = "is_deleted=false")
```
Comment (okuzan) [inline src/main/java/com/example/onlinebookstore/model/User.java]:
```suggestion
@SQLDelete(sql = "UPDATE employees SET is_deleted = TRUE WHERE id = ?")
@SQLRestriction(value = "is_deleted = FALSE")
```

**Review by okuzan — APPROVED** [review]

**Pull request merged**